File: src/TaskOTime/TaskOTime.ViewModel/ViewModels/TaskCompletionRequestEventArgs.vb
Namespace ViewModels
    Public Class TaskCompletionRequestEventArgs
        Inherits EventArgs
        Public Sub New(task As TaskItemViewModel, completedAt As DateTime, Optional useExistingBoundary As Boolean = False)
            Me.Task = task
            Me.CompletedAt = completedAt
            Me.UseExistingBoundary = useExistingBoundary
        End Sub
        Public ReadOnly Property Task As TaskItemViewModel
        Public ReadOnly Property CompletedAt As DateTime
        Public ReadOnly Property UseExistingBoundary As Boolean
    End Class
End Namespace

File: src/TaskOTime/TaskOTime.ViewModel/ViewModels/TaskCompletionRequestEventArgs.cs
using System;

namespace TaskOTime.ViewModel.ViewModels
{
    public class TaskCompletionRequestEventArgs : EventArgs
    {
        public TaskCompletionRequestEventArgs(TaskItemViewModel task, DateTime completedAt, bool useExistingBoundary = false)
        {
            Task = task;
            CompletedAt = completedAt;
            UseExistingBoundary = useExistingBoundary;
        }
        public TaskItemViewModel Task { get; private set; }
        public DateTime CompletedAt { get; private set; }
        public bool UseExistingBoundary { get; private set; }
        internal Action RollbackBooking { get; set; }
    }
}